src/signTx.c

```diff
@@ -133,42 +132,42 @@ static inline void advanceStage()
 
 	case SIGN_STAGE_BODY_TTL:
 		if (ctx->includeTtl) {
-			ASSERT(txBodyCtx->ttlReceived);
+			ASSERT(ACCESS_BODY_CTX()->ttlReceived);
 		}
 
 		ctx->stage = SIGN_STAGE_BODY_CERTIFICATES;
 
 		if (ctx->numCertificates > 0) {
-			txHashBuilder_enterCertificates(&txBodyCtx->txHashBuilder);
+			txHashBuilder_enterCertificates(&ACCESS_BODY_CTX()->txHashBuilder);
 			break;
 		}
 
 	// intentional fallthrough
 
 	case SIGN_STAGE_BODY_CERTIFICATES:
 		// we should have received all certificates
-		ASSERT(txBodyCtx->currentCertificate == ctx->numCertificates);
+		ASSERT(ACCESS_BODY_CTX()->currentCertificate == ctx->numCertificates);
 
 		ctx->stage = SIGN_STAGE_BODY_WITHDRAWALS;
 
 		if (ctx->numWithdrawals > 0) {
-			txHashBuilder_enterWithdrawals(&txBodyCtx->txHashBuilder);
+			txHashBuilder_enterWithdrawals(&ACCESS_BODY_CTX()->txHashBuilder);
 			break;
 		}
 
 	// intentional fallthrough
 
 	case SIGN_STAGE_BODY_WITHDRAWALS:
 		// we should have received all withdrawals
-		ASSERT(txBodyCtx->currentWithdrawal == ctx->numWithdrawals);
+		ASSERT(ACCESS_BODY_CTX()->currentWithdrawal == ctx->numWithdrawals);
 
 		if (ctx->includeAuxData) {
-			ASSERT(txAuxDataCtx->auxDataReceived);
+			// ASSERT(txAuxDataCtx->auxDataReceived);	TODO(KoMa) why is this check here? We are deep in txBody territory here!
```

Comment:
Seems like a bug, some leftover from the times where aux data were received at the end of tx body after withdrawals.

Reply:
Removed then.